### DISCUSSION TITLE
General idea behind Datagrid visualisation

### DISCUSSION THREAD

**Discussion by stavr666** [grafana/grafana#76468] [General]
While this Excel-like-for-children experience seems useful, right now I can't find any working scenario for it. Rendering data-set as editable table or copy-pasting from/to are good... But data have no value if you can't transform or visualize it.

There is potential usefulness. I can replace bunch of textboxes on some dashboard to single Datagrid. Or fancy-render input Data from some basic source before displaying merged result in other panels.

At least 2 changes must be done, IMHO:

1. **Allow some of data (rows, columns or singe cells) can be locked.**
Than I can have predictable key-value pairs for other visualizations and replace my bunch-of-Textbox dashboards.

2. **Data from Grid must be as useful, as Variables.**
Right now, I can extract data from Grid as --Dashboard-- datasource. This can't be used in Mixed source, so I can't extract any data and make custom requests (SQL, jsonata) and/or join them with other datasources. Only transformations available, limited to 4 arithmetic operations + whole-row aggregations.

Option №3 - add some calculation functions to Datagrid. Basic mathematical transformations for selected range of rows inside cell of Grid, without transformation tab. If change №2 is made, then №3 is generally not needed. But still useful for those just learning Grafana.

Related test board (Datagrid-based SLA calculator, used 6 transformations in Stat panel):
![image](https://github.com/grafana/grafana/assets/84839985/d2c9f5bf-450e-441c-ae33-6fe35f1b4544)

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move https://github.com/grafana/grafana/discussions/73424.

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!
